addons/mrp/models/mrp_production.py

```diff
@@ -1064,6 +1064,9 @@ def action_confirm(self):
     def action_assign(self):
         for production in self:
             production.move_raw_ids._action_assign()
+
+            # run scheduler for moves not fully reserved
+            production.move_raw_ids[0]._trigger_scheduler(production.move_raw_ids)
```

Comment:
You can remove the api.model and `moves` in the signature. and use 
``` python
for move in self:
    ....
```
Same idea for _trigger_assign. 

Also I would write this one like:
``` python
for production in self:
    production.move_raw_ids._action_assign()

# run scheduler for moves not fully reserved
self.move_raw_ids._trigger_scheduler()
``` 
to launch _procure_orderpoint_confirm() only once by company

Reply:
Ah of course. I'm going to say my brain is still tired from the barcode task as my excuse... It's now fixed.